x-pack/plugins/stack_connectors/public/connector_types/inference/get_task_types.ts

```diff
@@ -0,0 +1,607 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { DisplayType, FieldType } from '../lib/dynamic_config/types';
+import { FieldsConfiguration } from './types';
+
+export interface InferenceTaskType {
+  task_type: string;
+  configuration: FieldsConfiguration;
+}
+
+export const getTaskTypes = (http: HttpSetup, provider: string): Promise<InferenceTaskType[]> => {
```

Comment:
```suggestion
export const getTaskTypes = (provider: string): InferenceTaskType[] => {
```

`http` appears to be unused. Also no reason for async

Reply:
Also, consider renaming to `getTaskTypesByProvider`